### PR TITLE
Enhancements for Feed & Profile Page

### PR DESCRIPTION
**Pull request by ceylingebes**
This pull request introduces significant improvements and additions to the profile page and related functionalities, including the integration of backend endpoints, enhanced user profile editing, and password reset functionalities. Key features include dynamic profile updates, bookmark display, top contributors on the feed, and robust password recovery mechanisms.

### **List of Changes Made:**

1. **_Profile Page Enhancements:_** Added a new profile page for users to view their details, questions, and bookmarks. Displayed user's profile picture, username, and bio on the profile page. Implemented tabs for "Questions" and "Bookmarks." Added the ability for users to view only their bookmarks and questions, while removing these tabs on other users' profiles. Checked if the profile being viewed belongs to the logged-in user to control access to editing and other functionalities. Removed questions and bookmarks tabs for other users’ profiles. Added profile editing functionality via "Edit Profile" button.
Connected this functionality to the backend via the edit_user_profile endpoint. Added functionality to upload a profile picture using the upload_profile_pic endpoint. Handled backend integration to fetch the new profile picture URL and update the state.

2. **_Forgot Password and Reset Password:_**: Added a page where users can request a password reset by entering their email. Connected the form to the backend reset_password endpoint. Added a page where users can set a new password using a unique reset token. Integrated the update_password backend endpoint. Included password validation (matching "new password" and "confirm password").

3. _**Top Contributors on Feed:**_: Added a section on the feed page to display the top 5 contributors based on their activity and integrated the get_top_five_contributors backend endpoint.

4. **_Bug Fixes and Improvements:_**: Fixed the issue where the default profile picture disappeared after uploading a new picture. Updated the logic to dynamically reload the profile picture URL after an upload. Improved the display of top contributors on the feed page. Fixed a bug in the "delete user" function where the user ID comparison failed.

### **Files Changed:**

- Profile.js: Added profile page with enhanced functionalities.

- ForgotPassword.js: Added forgot password functionality.

- ResetPassword.js: Added reset password page.

- Feed.js: Updated feed to include top contributors.

- PageComponents.js: Updated Navbar with dynamic username caching.

### CODE REVIEW

**Comment by Mutti499** [comment]
I pulled your code to my branch and everything seems good howeverin the password reset the CSS file changing all the layout in the pages. It needs to be updated

**Review by Mutti499 — CHANGES_REQUESTED** [review]

**Comment by damlakayikci** [comment]
I just checked it and have some questions.
The "update password" button doesn't seem to work, it's not clickable for my case. Is it normal?
And I think it would look better if the navbar spans the whole page, now it looks like there is a gap in right and left sides of the page, like in this picture:
image

**Comment by ErayEroglu** [comment]
How are we doing on this PR, does anybody need help?

File: koduyorum-web/src/ResetPassword.css [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    margin: 50px auto;
+    padding: 20px;
+    background: #ffffff;
+    border: 1px solid #e0e0e0;
+    border-radius: 8px;
+    text-align: center;
+    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
+}
+
+h1 {
```
[the remark this answers is not in the document]
Comment (ceylingebes) [reply koduyorum-web/src/ResetPassword.css]: you're absolutely right, i fixed it on my recent push

**Comment by ceylingebes** [comment]
I've made the changes requested by @Mutti499 , can you review again? Also about the changes @damlakayikci mentioned, yes you're right, it would look better but I think navbar updates are not belong to this PR, and it is assigned to someone else on our team. If there is no PRs left to cover this, we can fix it on another branch maybe? It would just not related to profile changes right now in my opinion.

**Comment by ceylingebes** [comment]
I have resolved conflicts by accepting the changes in web-profile branch. However, keep in mind that this would also create conflict when trying to merge into main. I think we should take what is in main then.

**Comment by Mutti499** [comment]
If you are going to merge that to main you should be fast! Lets goo

**Pull request merged**